torchvision/prototype/datasets/_folder.py

```diff
@@ -0,0 +1,74 @@
+import io
+import os
+import os.path
+import pathlib
+from typing import Callable, Optional, Collection
+from typing import Union, Tuple, List, Dict, Any
+
+import torch
+from torch.utils.data import IterDataPipe
+from torch.utils.data.datapipes.iter import FileLister, Concater, FileLoader, Mapper, Shuffler
+
+from torchvision.prototype.datasets.decoder import pil
+
+
+__all__ = ["from_data_folder", "from_image_folder"]
+
+# pseudo-infinite buffer size until a true infinite buffer is supported
+INFINITE = 1_000_000_000
```

Comment:
@VitalyFedyunin @ejguan We need the ability to specify an infinite buffer, i.e. `buffer_size=None` for every datapipe that implements a buffer.